### ISSUE TITLE
Add proper tests for the change to the module loader in astroid

### ISSUE THREAD

**Issue by Pierre-Sassoulas** [PyCQA/pylint#5341]
See https://github.com/PyCQA/astroid/pull/1207#issuecomment-951455588

I'm wondering it it's mergeable without the test added in pylint first. It looks like a valuable MR but it's also hard to test. What do you think @DanielNoord @cdce8p ? Should we merge in 2.8.6 test in pylint after 2.12 ? Or do not merge in astroid until there are regression tests in pylint ?

**Comment by DanielNoord** [comment]
I have not followed the discussion but the test given (I think) can be ported to `astroid`:

```python
    def test_safe_infer_on_bz2_compress_return(self):
        src_node = astroid.extract_node('import bz2\nunused = bz2.compress(b"")')
        assignment = next(src_node.nodes_of_class(astroid.Assign))
        inferred = pylint.checkers.utils.safe_infer(assignment.value.func)
        self.assertIsInstance(inferred, astroid.FunctionDef, msg=str(inferred))
        try:
            next(inferred.nodes_of_class(astroid.Return))
        except StopIteration:
            self.fail(f'No return found in body of:\n{str(inferred)}.')
```
I think we have a `safe_infer` in `astroid`. That means that we could also add a single test on namespace packages in the `astroid` test suite. We could probably borrow some code from how the primer is being set up. Would that work?
